File: Selection_Sort.py
from random import randint

def smallest(list):
    small = list[0]
    small_index = 0
    for i in range(1,len(list)):
        if list[i] < small:
            small = list[i]
            small_index = i
    return small_index

def selectionsort(list):
    arr = []
    for i in range(len(list)):
        small = smallest(list)
        arr.append(list.pop(small))
    return arr

def binary_search(list,to_search):
    low = 0
    high = len(list) - 1
    mid = 0

    while low <= high:
        mid = (high + low) // 2

        if list[mid] < to_search:
            low = mid + 1
        elif list[mid] > to_search:
            high = mid - 1
        else:
            return mid
    return -1


if __name__ == '__main__':
    numbers = int(input("Enter Array Size : "))
    list = []
    for i in range(numbers):
        list.append(randint(0,100))
    list = selectionsort(list)
    for i in list:
        print(i,end="   ")
    print()
    to_search = int(input("Enter Number to Search : "))
    result = binary_search(list,to_search)
    print(to_search , " fount at position : " , result)

File: Selection_Sort.cpp
#include <bits/stdc++.h>
using namespace std;

int binarySearch(int arr[], int to_search, int size)
{
  int high,low,mid;
  low = 0;
  high = size;
  while (low <= high)
  {
    mid = (high+low)/2;
    if (arr[mid] < to_search)
    {
      low = mid+1;
    }
    else if (arr[mid] > to_search)
    {
      high = mid-1;
    }
    else
      return mid;
  }
  return -1;
}

void swap(int & small, int & large)
{
  int k = large;
  large = small;
  small = k;
}

void sort(int arr[], int n)
{
  for(int i = 0; i < n-1; i++)
  {
    int small = i;
    for(int j = i+1; j < n; j++)
    {
      if(arr[small] > arr[j])
        swap(arr[small], arr[j]);
    }
  }
}

int main(int argc, char const *argv[]) {
  int n;
  cout << "Enter Number Of Elements : ";
  cin >> n;
  int arr[n];
  srand(time(0));
  for (int i = 0; i < n; i++)
  {
    arr[i] = rand()%100;
  }
  //sort(arr,arr+n);
  sort(arr,n);
  for (int i = 0; i < n; i++)
  {
    cout << arr[i] << " ";
  }
  cout << "\nEnter Element To Search : ";
  int to_search;
  cin >> to_search;
  int pos;
  pos = binarySearch(arr, to_search, n);
  cout << to_search << " found at position : " << pos << endl;
  return 0;
}
